### DISCUSSION TITLE
Working with Blender

### DISCUSSION THREAD

**Discussion by keitaro818** [coldtype/coldtype#114] [Q&A]
Hello.
I recently found out about "coldtype". Thank you for providing such a great tool.

I am currently testing the integration with Blender, but I keep getting the following error.
If you know the cause, please let me know.

```
location: <unknown location>:-1
Error: Python: Traceback (most recent call last):
  File "/Applications/Blender.app/Contents/Resources/2.93/python/lib/python3.9/site-packages/coldtype/blender/watch.py", line 316, in modal
    start, kwargs = [t.strip() for t in line.split(";")]
ValueError: not enough values to unpack (expected 2, got 1)

location: <unknown location>:-1
```

**Comment by stenson** [comment]
Sorry about that error! It looks like the coldtype/blender is broken in Blender 2.93 unfortunately, mostly because I haven't used 2.93 in a while. I'll look into fixing it, but because of some issues with the python bundled in Blender 2.93, it's become a lot easier to install and test the coldtype/blender integration with Blender v3.0 and newer (I'm currently mainly using Blender 3.2.0). If you're using 2.93 for older projects and you need to keep it around, you can install a newer 3.0+ Blender alongside the older one and they should both work (i.e. rather than drag the Blender app into your Applications directory, you can drag it to your Desktop or anywhere else and just rename the app to Blender3.2 or something like that, I've got a whole directory full of them)

<img width="328" alt="image" src="https://user-images.githubusercontent.com/89770/179030785-6c67d5cd-8f41-4d79-8620-7f95eee2165c.png">

**Reply by keitaro818** [reply]
Thank you for your response.
I immediately upgraded Blender to the latest version, but the same error seems to occur.
It works up to the point where Blender starts up, but then I get this error.
```
Error: Python: Traceback (most recent call last):
  File "/Applications/Blender.app/Contents/Resources/3.2/python/lib/python3.10/site-packages/coldtype/blender/watch.py", line 316, in modal
    start, kwargs = [t.strip() for t in line.split(";")]
ValueError: not enough values to unpack (expected 2, got 1)
```

The environment is as follows
- macOS Big sur 11.5.2
- Blender 3.2.1
- python 3.10

**Reply by stenson** [reply]
Hi @keitaro818, sorry for the (very long) delay! I did try to dig into the error but never made much headway, and to be perfectly honest the coldtype->blender pipeline has always been pretty fragile. That's part of the reason why I started work (around when you filed this issue) on [ST2](https://github.com/coldtype/st2), which is a GUI frontend to Coldtype built as a Blender addon. Just posted some videos about it (here's one about [the backstory](https://www.youtube.com/watch?v=gV2laWd727U)), but you might be more interested in [Scripting for ST2](https://www.youtube.com/watch?v=TrZ6tIde2P0), which enables a lot of the same functionality as the coldtype->blender pipeline you tried out here.

I will still try to fix this issue, just haven't had any luck recreating it on my setup.

**Reply by keitaro818** [reply]
Hello @stenson, thank you for your answer.
I had already started using Blender ST2. I think it's a very nice add-on!
I also watched your video. Thank you very much. By the way, is it possible to animate by script?

**Reply by stenson** [reply]
Good to hear! Not yet possible to script an animation, but I am planning on support for scripted animation in ST2, hopefully pretty soon!

**Reply by keitaro818** [reply]
It is very wonderful. Looking forward to it!
